src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -1133,6 +1060,88 @@ private StateEditor doTraverse(
     return s1;
   }
 
+  private double getReluctance(RoutingRequest req, TraverseMode traverseMode, boolean walkingBike) {
+    if (isStairs() && req.wheelchairAccessibility.enabled()) {
+      return req.wheelchairAccessibility.stairsReluctance();
+    } else if (isStairs()) {
+      return req.stairsReluctance;
+    } else {
+      return req.getReluctance(traverseMode, walkingBike);
+    }
+  }
```

Comment:
Here important business logic is pushed to the Request object - a DTO. A better way is to define an interface for the needed parameters that the request can provide/implement and put the logic in a "Calculator".

Reply:
I actually didn't write this logic but I can look into it anyway.

Reply:
This should be moved into it's own interface like `BikeWalkableEdge`.

Reply:
I've introduced a new interface `ReluctanceEdge` to compute the reluctance in StreetEdge and PathwayEdge.